index.js

```diff
@@ -78,12 +82,48 @@ function runNightwatch (done) {
 			} else if (argv.exclude_paths) {
 				process.env.KNE_EXCLUDE_TEST_PATHS += ',' + argv.exclude_paths;
 			}
-			process.env.KNE_SELENIUM_START_PROCESS = process.env.KNE_SELENIUM_START_PROCESS || argv.selenium_start_process || true;
 
 			argv.config = path.resolve(__dirname, 'nightwatch.conf.js');
-			Nightwatch.runner(argv, function () {
-				console.log([moment().format('HH:mm:ss:SSS')] + ' kne: finished tests...');
-				done();
+
+			async.series([
+
+				/**
+				 * If the user wants us to start selenium manually then do so
+				 */
+				function (cb) {
+					if (argv.env === 'default' && argv['selenium-in-background']) {
+						runSeleniumInBackground(cb);
+					} else {
+						cb();
+					}
+				},
+
+				/**
+				 * The only environment that currently requires starting sauce connect is travis.
+				 */
+				function (cb) {
+					if (argv.env === 'saucelabs-travis') {
+						startSauceConnect(cb);
+					} else {
+						cb();
+					}
+				},
+
+				/**
+				 * Run nightwatch to start executing the tests
+				 */
+				function (cb) {
+					Nightwatch.runner(argv, function (status) {
+						if (status) {
+							console.log([moment().format('HH:mm:ss:SSS')] + ' kne: tests passed');
+						} else {
+							console.log([moment().format('HH:mm:ss:SSS')] + ' kne: tests failed');
+						}
+						cb();
+					});
```

Comment:
@webteckie You're right, closing sauce connect happens below. I was getting confused, I think.